From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Debayan Kar <kardebayan3@gmail.com>
Date: Sun, 25 Oct 2026 16:07:22 +0000
Subject: [PATCH 12/12] GeneralUtils: Drop MIUI window and notification
 extensions

Window.addExtraFlags() and Notification.extraNotification only exist in
the MIUI framework. The street snap notification would crash with a
NoSuchFieldError when posted, so stop setting the MIUI message count.
Also make the unused notch flag helper a no-op.
---
 smali/com/android/camera/GeneralUtils.smali | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/smali/com/android/camera/GeneralUtils.smali b/smali/com/android/camera/GeneralUtils.smali
index 58956ba4..a678d78f 100644
--- a/smali/com/android/camera/GeneralUtils.smali
+++ b/smali/com/android/camera/GeneralUtils.smali
@@ -17,11 +17,7 @@
 .end method
 
 .method public static applyNotchFlag(Landroid/view/Window;)V
-    .locals 1
-
-    const/16 v0, 0x700
-
-    invoke-virtual {p0, v0}, Landroid/view/Window;->addExtraFlags(I)V
+    .locals 0
 
     return-void
 .end method
@@ -217,10 +213,6 @@
 
     iput p2, p0, Landroid/app/Notification;->flags:I
 
-    iget-object p2, p0, Landroid/app/Notification;->extraNotification:Landroid/app/MiuiNotification;
-
-    invoke-virtual {p2, p1}, Landroid/app/MiuiNotification;->setMessageCount(I)V
-
     invoke-virtual {p3, p1, p0}, Landroid/app/NotificationManager;->notify(ILandroid/app/Notification;)V
     :try_end_0
     .catch Ljava/lang/NullPointerException; {:try_start_0 .. :try_end_0} :catch_0
-- 
2.43.0

